cmd/snap-update-ns/mount-entry-change-test.c

```diff
@@ -0,0 +1,232 @@
+	sc_test_write_lines("desired.fstab",
+			    test_entry_str_1, test_entry_str_2, NULL);
+
+	current = sc_load_mount_profile("current.fstab");
+	desired = sc_load_mount_profile("desired.fstab");
+	g_test_queue_destroy((GDestroyNotify) sc_free_mount_entry_list,
+			     current);
+	g_test_queue_destroy((GDestroyNotify) sc_free_mount_entry_list,
+			     desired);
+
+	change = sc_compute_required_mount_changes(desired, current);
+	g_test_queue_destroy((GDestroyNotify) sc_mount_change_free_chain,
+			     change);
+
+	test_assert_change_list(change,
+				&test_entry_1, SC_ACTION_MOUNT,
+				&test_entry_2, SC_ACTION_MOUNT, NULL);
+}
+
+// Scenario: the current profile contains one entry but the desired profile
+// contains two entries. We should see one mount change (for the 2nd entry).
+static void test_sc_compute_required_mount_changes__scenario3()
+{
+	struct sc_mount_entry *current;
+	struct sc_mount_entry *desired;
+	struct sc_mount_change *change;
+
+	sc_test_write_lines("current.fstab", test_entry_str_1, NULL);
+	sc_test_write_lines("desired.fstab",
+			    test_entry_str_1, test_entry_str_2, NULL);
+
+	current = sc_load_mount_profile("current.fstab");
+	desired = sc_load_mount_profile("desired.fstab");
+	g_test_queue_destroy((GDestroyNotify) sc_free_mount_entry_list,
+			     current);
+	g_test_queue_destroy((GDestroyNotify) sc_free_mount_entry_list,
+			     desired);
+
+	change = sc_compute_required_mount_changes(desired, current);
+	g_test_queue_destroy((GDestroyNotify) sc_mount_change_free_chain,
+			     change);
+
+	test_assert_change_list(change, &test_entry_2, SC_ACTION_MOUNT, NULL);
+}
+
+// Scenario: the current profile contains one entry and the desired profile
+// contains one entry but they are different. We should see the unmount
+// followed by the mount.
+static void test_sc_compute_required_mount_changes__scenario4()
+{
+	struct sc_mount_entry *current;
+	struct sc_mount_entry *desired;
+	struct sc_mount_change *change;
+
+	sc_test_write_lines("current.fstab", test_entry_str_1, NULL);
+	sc_test_write_lines("desired.fstab", test_entry_str_2, NULL);
+
+	current = sc_load_mount_profile("current.fstab");
+	desired = sc_load_mount_profile("desired.fstab");
+	g_test_queue_destroy((GDestroyNotify) sc_free_mount_entry_list,
+			     current);
+	g_test_queue_destroy((GDestroyNotify) sc_free_mount_entry_list,
+			     desired);
+
+	change = sc_compute_required_mount_changes(desired, current);
+	g_test_queue_destroy((GDestroyNotify) sc_mount_change_free_chain,
+			     change);
+
+	test_assert_change_list(change,
+				&test_entry_1, SC_ACTION_UNMOUNT,
+				&test_entry_2, SC_ACTION_MOUNT, NULL);
+}
+
+// Scenario: desired A, B current B, C behaves correctly (B is untouched).
+static void test_sc_compute_required_mount_changes__scenario5()
+{
+	struct sc_mount_entry *current;
+	struct sc_mount_entry *desired;
+	struct sc_mount_change *change;
+
+	sc_test_write_lines("desired.fstab", "A A A A 0 0", "B B B B 0 0",
```

Comment:
Wonder if we can have a `test_entry_3` here? I found it slightly confusing why this is different than the other tests.